src/Microsoft.Azure.WebJobs.Host/Loggers/TraceEventExtensions.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Azure.WebJobs.Host.Protocols;
+
+namespace Microsoft.Azure.WebJobs.Host
+{
+    internal static class TraceEventExtensions
+    {
+        private const string FunctionDescriptorKey = "MS_FunctionDescriptor";
+        private const string FunctionInvocationIdKey = "MS_FunctionInvocationId";
+        private const string HostIdKey = "MS_HostId";
+
+        public static void AddFunctionInstanceDetails(this TraceEvent traceEvent, Guid hostId, FunctionDescriptor descriptor, Guid functionId)
+        {
+            traceEvent.Properties[HostIdKey] = hostId;
+            traceEvent.Properties[FunctionDescriptorKey] = descriptor;
```

Comment:
Yeah it's public. It includes a few other things that someone may want -- I didn't have a good reason to restrict them: https://github.com/Azure/azure-webjobs-sdk/blob/master/src/Microsoft.Azure.WebJobs.Protocols/FunctionDescriptor.cs